internal/pkg/agent/cmd/unprivileged.go

```diff
@@ -0,0 +1,127 @@
+)
+
+const (
+	endpoint = "endpoint"
+)
+
+func newUnprivilegedCommandWithArgs(s []string, streams *cli.IOStreams) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "unprivileged",
+		Short: "Switch installed Elastic Agent to run as unprivileged",
+		Long: `This command converts the installed Elastic Agent from running privileged to running as unprivileged.
+
+By default this command will ask or a confirmation before making this change. You can bypass the confirmation request
+using the -f flag. This will always stop the running Elastic Agent (if running) before performing the switch it is
+possible that loss of metrics could occur during this small window of time. The command also always starts the
+Elastic Agent at the end (even if it was off to start). In the case that the Elastic Agent is already running
+unprivileged it will still perform all the same work, including stopping and starting the Elastic Agent.
+`,
+		Args: cobra.ExactArgs(0),
+		Run: func(c *cobra.Command, args []string) {
+			if err := unprivilegedCmd(streams, c); err != nil {
+				fmt.Fprintf(streams.Err, "Error: %v\n%s\n", err, troubleshootMessage())
+				os.Exit(1)
+			}
+		},
+	}
+
+	cmd.Flags().BoolP("force", "f", false, "Do not prompt for confirmation")
+	cmd.Flags().DurationP("daemon-timeout", "", 0, "Timeout waiting for Elastic Agent daemon")
+
+	return cmd
+}
+
+func unprivilegedCmd(streams *cli.IOStreams, cmd *cobra.Command) (err error) {
+	isAdmin, err := utils.HasRoot()
+	if err != nil {
+		return fmt.Errorf("unable to perform unprivileged command while checking for root/Administrator rights: %w", err)
+	}
+	if !isAdmin {
+		return fmt.Errorf("unable to perform unprivileged command, not executed with %s permissions", utils.PermissionUser)
+	}
+
+	// TODO(blakerouse): More work to get this working on macOS.
+	// Need to switch the vault from keystore based to file based vault.
+	if runtime.GOOS == "darwin" {
+		return errors.New("unable to perform unprivileged on macOS (not supported)")
+	}
+
+	// cannot switch to unprivileged when Elastic Defend exists in the policy
+	err = ensureNoElasticDefend()
+	if err != nil {
+		// error already adds context
+		return err
+	}
+
+	topPath := paths.Top()
+	daemonTimeout, _ := cmd.Flags().GetDuration("daemon-timeout")
+	force, _ := cmd.Flags().GetBool("force")
+	if !force {
+		confirm, err := cli.Confirm("This will restart the running Elastic Agent and convert it to run in unprivileged mode. Do you want to continue?", true)
+		if err != nil {
+			return fmt.Errorf("problem reading prompt response")
+		}
+		if !confirm {
+			return fmt.Errorf("unprivileged switch was cancelled by the user")
+		}
+	}
+
+	pt := install.CreateAndStartNewSpinner(streams.Out, "Converting Elastic Agent to unprivileged...")
+	err = install.SwitchExecutingMode(topPath, pt, install.ElasticUsername, install.ElasticGroupName)
+	if err != nil {
+		// error already adds context
+		return err
+	}
+
+	// wait for the service
+	if daemonTimeout >= 0 {
```

Comment:
Same question as https://github.com/elastic/elastic-agent/pull/4621/files?diff=split&w=1#r1603913060

Reply:
Yes because they can provide a negative number which means to not wait.